src/data/equiv/encodable.lean

```diff
@@ -289,20 +290,67 @@ theorem skolem {α : Type*} {β : α → Type*} {P : Π x, β x → Prop}
 
 end encodable
 
-namespace quot
+namespace directed
+
 open encodable
+
+variables {α : Type*} {β : Type*} [encodable α] [inhabited α]
+
+/-- Given a `directed r` function `f : α → β` defined on an encodable inhabited type,
+construct a noncomputable sequence such that `r (f (x n)) (f (x (n + 1)))`
+and `r (f a) (f (x (encode a + 1))`. -/
+protected noncomputable def sequence {r : β → β → Prop} (f : α → β) (hf : directed r f) : ℕ → α
```

Comment:
Shouldn't `f` be implicit?

Reply:
Will it lead to `x = directed.sequence _` in the output?

Reply:
Yup...

Reply:
OK, it's a good argument to keep it like that.